QUICK_START_GUIDE.md

```diff
@@ -11,175 +11,286 @@
+       --extra-index-url https://download.pytorch.org/whl/rocm6.0 \
+       -C cmake.args="-DLLAMA_HIPBLAS=on" \
+       -C cmake.args="-DAMDGPU_TARGETS=all" \
+       -C cmake.args="-DCMAKE_C_COMPILER=/opt/rocm/llvm/bin/clang" \
+       -C cmake.args="-DCMAKE_CXX_COMPILER=/opt/rocm/llvm/bin/clang++" \
+       -C cmake.args="-DCMAKE_PREFIX_PATH=/opt/rocm"
    ```
 
-   > **NOTE**: ⏳ `pip install` may take some time, depending on your internet connection.
+   On Fedora 40+, use `-DCMAKE_C_COMPILER=clang-17` and `-DCMAKE_CXX_COMPILER=clang++-17`.
+
+   #### To install with Apple Metal on M1/M2/M3 Mac
+
+   ```shell
+   python3 -m venv --upgrade-deps venv
+   source venv/bin/activate
+   (venv) $ pip cache remove llama_cpp_python
+   (venv) $ pip install git+https://github.com/instructlab/instructlab.git@stable -C cmake.args="-DLLAMA_METAL=on"
+   ```
+
+   #### To install with Nvidia CUDA
+
+   ```shell
+   python3 -m venv --upgrade-deps venv
+   source venv/bin/activate
+   (venv) $ pip cache remove llama_cpp_python
+   (venv) $ pip install git+https://github.com/instructlab/instructlab.git@stable -C cmake.args="-DLLAMA_CUBLAS=on"
+   ```
 
 4. From your `venv` environment, verify `ilab` is installed correctly, by running the `ilab` command.
 
-   ```bash
+   ```shell
    ilab
    ```
 
-#### Example output
+   #### Example output
 
    ```shell
    (venv) $ ilab
    Usage: ilab [OPTIONS] COMMAND [ARGS]...
 
    CLI for interacting with InstructLab.
 
-   If this is your first time running `ilab`, it's best to start with `ilab init`
-   to create the environment
+   If this is your first time running InstructLab, it's best to start with `ilab init` to create the environment.
 
    Options:
    --config PATH  Path to a configuration file.  [default: config.yaml]
+   --version      Show the version and exit.
    --help         Show this message and exit.
 
    Commands:
    chat      Run a chat using the modified model
-   check     Check that taxonomy is valid
+   check     (Deprecated) Check that taxonomy is valid
    convert   Converts model to GGUF
+   diff      Lists taxonomy files that have changed since <taxonomy-base>...
    download  Download the model(s) to train
    generate  Generates synthetic data to enhance your example data
    init      Initializes environment for InstructLab
-   list      Lists taxonomy files that have changed since a reference commit (default origin/main)
+   list      (Deprecated) Lists taxonomy files that have changed since <taxonomy-base>.
    serve     Start a local server
    test      Runs basic test to ensure model correctness
    train     Takes synthetic data generated locally with `ilab generate`...
    ```
 
    > **IMPORTANT:** every `ilab` command needs to be run from within your Python virtual environment. To enter the Python environment, run the following command:
 
-   ```bash
+   ```shell
    source venv/bin/activate
    ```
 
+5. You may optionally enable tab completion for the `ilab` command.
```

Comment:
I strongly suggest not having this in this document. Adding this to a lab/workshop document will only confuse students.